### DISCUSSION TITLE
Automatically splitting text into multiple dialogue boxes

### DISCUSSION THREAD

**Discussion by JoelGalaxy** [dialogic-godot/dialogic#1414] [Q&A]
Hi, I'm fairly new to Godot and Dialogic for that matter so any help would be greatly appreciated, would it be possible to make it so Dialogic would only display as many characters as fit in the dialogue box, and then prompt with the 'next dialogue page' icon and await input, and then clear the text box and proceed to display the remainder of the full text? 

As it is now I can only make it so once the text doesn't fit all at once in the dialogue box, it scrolls to keep displaying the text. And although I could theoretically keep count of how many characters fit in the dialogue box, and separate manually the text into multiple text events in dialogic's timeline, it would be much easier if I could just type all the text said by one character at once and have dialogic or godot split it up, as I feel many dialogue systems in other games take this approach instead of displaying all of it scrolling.

Again any help is much appreciated!

**Comment by coppolaemilio** [comment]
Hello @JoelGalaxy. This is something that we all wanted at some point from Dialogic, but it's a really hard problem to solve. I'm sure there will be ways of making it a reality but it wasn't a priority yet.

There are many different measurements to calculate when auto-splitting text into different dialogs from a single text event. The font size, the width of each character (it's easier if the fonts are mono-spaced, but for visual novels most fonts are not), the container, where to split text (by word, sentece, syllable), etc...

I hope that once we have everything done in Dialogic 2 we can revisit this problem and see if there is a realistic solution to this. Looking at the bright side of it, by splitting the text manually, it can help you a bit with the pacing of the dialogs and keeping them shorter and with more intent.

Hope that answers your question :)
